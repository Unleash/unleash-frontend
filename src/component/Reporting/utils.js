import parseISO from 'date-fns/parseISO';
import differenceInDays from 'date-fns/differenceInDays';

import { EXPERIMENT, OPERATIONAL, RELEASE, FOURTYDAYS, SEVENDAYS } from './constants';

export const toggleExpiryByTypeMap = {
    [EXPERIMENT]: FOURTYDAYS,
    [RELEASE]: FOURTYDAYS,
    [OPERATIONAL]: SEVENDAYS,
};

export const applyCheckedToFeatures = (features, checkedState) =>
    features.map(feature => ({ ...feature, checked: checkedState }));

export const getCheckedState = (name, features) => {
    const feature = features.find(feature => feature.name === name);

    if (feature) {
        return feature.checked ? feature.checked : false;
    }
    return false;
};

export const getDiffInDays = (date, now) => Math.abs(differenceInDays(date, now));

export const formatProjectOptions = projects => projects.map(project => ({ key: project.id, label: project.name }));

export const expired = (diff, type) => {
    if (diff >= toggleExpiryByTypeMap[type]) return true;
    return false;
};

export const getObjectProperties = (target, ...keys) => {
    const newObject = {};

    keys.forEach(key => {
        if (target[key] !== undefined) {
            newObject[key] = target[key];
        }
    });

    return newObject;
};

export const sortFeaturesByNameAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.name < b.name) {
            return -1;
        }
        if (a.name > b.name) {
            return 1;
        }
        return 0;
    });
    return sorted;
};

export const sortFeaturesByNameDescending = features => sortFeaturesByNameAscending([...features]).reverse();

export const sortFeaturesByLastSeenAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (!a.lastSeenAt) return -1;
        if (!b.lastSeenAt) return 1;

        const dateA = parseISO(a.lastSeenAt);
        const dateB = parseISO(b.lastSeenAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByLastSeenDescending = features => sortFeaturesByLastSeenAscending([...features]).reverse();

export const sortFeaturesByCreatedAtAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByCreatedAtDescending = features => sortFeaturesByCreatedAtAscending([...features]).reverse();

export const sortFeaturesByExpiredAtAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const now = new Date();
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type)) return -1;
        if (!expired(diffB, b.type)) return 1;

        console.log('EXPIRED');

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByA - expiredByB;
    });
    return sorted;
};

export const sortFeaturesByExpiredAtDescending = features => sortFeaturesByExpiredAtAscending([...features]).reverse();

export const sortFeaturesByStatusAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.stale) return 1;
        if (b.stale) return -1;
        return 0;
    });
    return sorted;
};

export const sortFeaturesByStatusDescending = features => sortFeaturesByStatusAscending([...features]).reverse();

export const pluralize = (items, word) => {
    if (items === 1) return `${items} ${word}`;
    return `${items} ${word}s`;
};

export const getDates = dateString => {
    const date = parseISO(dateString);
    const now = new Date();

    return [date, now];
};

export const filterByProject = selectedProject => feature => feature.project === selectedProject;

export const isFeatureExpired = feature => {
    const [date, now] = getDates(feature.createdAt);
    const diff = getDiffInDays(date, now);

    return expired(diff, feature.type);
};
