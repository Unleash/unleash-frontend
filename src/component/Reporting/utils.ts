import parseISO from 'date-fns/parseISO';
import differenceInDays from 'date-fns/differenceInDays';

import { EXPERIMENT, OPERATIONAL, RELEASE } from 'constants/featureToggleTypes';

import { FOURTYDAYS, SEVENDAYS } from './constants';
import { IFeatureToggleListItem } from 'interfaces/featureToggle';

export const toggleExpiryByTypeMap: Record<string, number> = {
    [EXPERIMENT]: FOURTYDAYS,
    [RELEASE]: FOURTYDAYS,
    [OPERATIONAL]: SEVENDAYS,
};

export interface IFeatureToggleListItemCheck extends IFeatureToggleListItem {
    checked: boolean;
}

export const applyCheckedToFeatures = (
    features: IFeatureToggleListItem[],
    checkedState: boolean
): IFeatureToggleListItemCheck[] => {
    return features.map(feature => ({
        ...feature,
        checked: checkedState,
    }));
};

export const getCheckedState = (
    name: string,
    features: IFeatureToggleListItemCheck[]
) => {
    const feature = features.find(feature => feature.name === name);

    if (feature) {
        return feature.checked;
    }

    return false;
};

export const getDiffInDays = (date: Date, now: Date) =>
    Math.abs(differenceInDays(date, now));

export const expired = (diff: number, type: string) => {
    if (diff >= toggleExpiryByTypeMap[type]) return true;
    return false;
};

export const getObjectProperties = <T extends object>(
    target: T,
    ...keys: (keyof T)[]
): Partial<T> => {
    const newObject: Partial<T> = {};

    keys.forEach(key => {
        if (target[key] !== undefined) {
            newObject[key] = target[key];
        }
    });

    return newObject;
};

export const sortFeaturesByNameAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.name < b.name) {
            return -1;
        }
        if (a.name > b.name) {
            return 1;
        }
        return 0;
    });
    return sorted;
};

export const sortFeaturesByNameDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByNameAscending([...features]).reverse();

export const sortFeaturesByLastSeenAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (!a.lastSeenAt) return -1;
        if (!b.lastSeenAt) return 1;

        const dateA = parseISO(a.lastSeenAt);
        const dateB = parseISO(b.lastSeenAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByLastSeenDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByLastSeenAscending([...features]).reverse();

export const sortFeaturesByCreatedAtAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByCreatedAtDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByCreatedAtAscending([...features]).reverse();

export const sortFeaturesByExpiredAtAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const now = new Date();
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type) && expired(diffB, b.type)) {
            return 1;
        }

        if (expired(diffA, a.type) && !expired(diffB, b.type)) {
            return -1;
        }

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByB - expiredByA;
    });
    return sorted;
};

export const sortFeaturesByExpiredAtDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    const now = new Date();
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type) && expired(diffB, b.type)) {
            return 1;
        }

        if (expired(diffA, a.type) && !expired(diffB, b.type)) {
            return -1;
        }

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByA - expiredByB;
    });
    return sorted;
};

export const sortFeaturesByStatusAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.stale) return 1;
        if (b.stale) return -1;
        return 0;
    });
    return sorted;
};

export const sortFeaturesByStatusDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByStatusAscending([...features]).reverse();

export const pluralize = (items: number, word: string): string => {
    if (items === 1) return `${items} ${word}`;
    return `${items} ${word}s`;
};

export const getDates = (dateString: string): [Date, Date] => {
    const date = parseISO(dateString);
    const now = new Date();

    return [date, now];
};
