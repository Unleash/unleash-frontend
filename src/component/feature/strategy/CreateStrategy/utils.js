export const resolveDefaultParamValue = (name, featureToggleName) => {
    switch (name) {
        case "percentage":
        case "rollout":
            return "100";
        case "stickiness":
            return "default";
        case "groupId":
            return featureToggleName;
        default:
            return "";
    }
};
