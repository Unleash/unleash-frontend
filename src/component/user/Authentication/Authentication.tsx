import SimpleAuth from '../SimpleAuth/SimpleAuth';
import AuthenticationCustomComponent from '../authentication-custom-component';
import PasswordAuth from '../PasswordAuth/PasswordAuth';
import HostedAuth from '../HostedAuth/HostedAuth';
import DemoAuth from '../DemoAuth/DemoAuth';
import {
    SIMPLE_TYPE,
    DEMO_TYPE,
    PASSWORD_TYPE,
    HOSTED_TYPE,
} from '../../../constants/authTypes';
import SecondaryLoginActions from '../common/SecondaryLoginActions/SecondaryLoginActions';
import { useAuth } from '../../../hooks/api/getters/useAuth/useAuth';
import useQueryParams from '../../../hooks/useQueryParams';
import ConditionallyRender from '../../common/ConditionallyRender';
import { Alert } from '@material-ui/lab';

const Authentication = () => {
    const { authDetails } = useAuth().auth ?? {};
    const params = useQueryParams();

    const error = params.get('errorMsg');
    if (!authDetails) return null;

    let content;
    if (authDetails.type === PASSWORD_TYPE) {
        content = (
            <>
                <PasswordAuth
                    authDetails={authDetails}
                />
                <ConditionallyRender
                    condition={!authDetails.defaultHidden}
                    show={<SecondaryLoginActions />}
                />
            </>
        );
    } else if (authDetails.type === SIMPLE_TYPE) {
        content = (
            <SimpleAuth
                authDetails={authDetails}
            />
        );
    } else if (authDetails.type === DEMO_TYPE) {
        content = (
            <DemoAuth
                authDetails={authDetails}
            />
        );
    } else if (authDetails.type === HOSTED_TYPE) {
        content = (
            <>
                <HostedAuth
                    authDetails={authDetails}
                />
                <ConditionallyRender
                    condition={!authDetails.defaultHidden}
                    show={<SecondaryLoginActions />}
                />
            </>
        );
    } else {
        content = <AuthenticationCustomComponent authDetails={authDetails} />;
    }
    return (
        <>
            <div style={{ maxWidth: '350px' }}>
                <ConditionallyRender
                    condition={Boolean(error)}
                    show={<Alert severity="error">{error}</Alert>}
                />
            </div>
            {content}
        </>
    );
};

export default Authentication;
