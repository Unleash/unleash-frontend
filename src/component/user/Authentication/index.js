import { connect } from 'react-redux';
import AuthenticationComponent from './Authentication';
import {
    insecureLogin,
    passwordLogin,
    demoLogin,
} from '../../../store/user/actions';

const mapDispatchToProps = (dispatch, props) => ({
    demoLogin: (path, user) => demoLogin(path, user)(dispatch),
    insecureLogin: (path, user) => insecureLogin(path, user)(dispatch),
    passwordLogin: (path, user) => passwordLogin(path, user)(dispatch),
});

export default connect(null, mapDispatchToProps)(AuthenticationComponent);
