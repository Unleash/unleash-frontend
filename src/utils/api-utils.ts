export const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
};

export class AuthenticationError extends Error {
    constructor(statusCode, body) {
        super('Authentication required');
        this.name = 'AuthenticationError';
        this.statusCode = statusCode;
        this.body = body;
    }
}

export class ForbiddenError extends Error {
    constructor(statusCode, body = {}) {
        super(
            body.details?.length > 0
                ? body.details[0].message
                : 'You cannot perform this action'
        );
        this.name = 'ForbiddenError';
        this.statusCode = statusCode;
        this.body = body;
    }
}

export class BadRequestError extends Error {
    constructor(statusCode, body = {}) {
        super(
            body.details?.length > 0 ? body.details[0].message : 'Bad request'
        );
        this.name = 'BadRequestError';
        this.statusCode = statusCode;
        this.body = body;
    }
}

export class NotFoundError extends Error {
    constructor(statusCode) {
        super(
            'The requested resource could not be found but may be available in the future'
        );
        this.name = 'NotFoundError';
        this.statusCode = statusCode;
    }
}
