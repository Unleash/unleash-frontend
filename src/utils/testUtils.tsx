import { SWRConfig } from 'swr';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import theme from 'themes/mainTheme';
import React from 'react';
import { SetupServerApi } from 'msw/node';
import { rest } from 'msw';

export const TestContext: React.FC = ({ children }) => {
    return (
        <SWRConfig value={{ provider: () => new Map() }}>
            <MemoryRouter>
                <ThemeProvider theme={theme}>{children}</ThemeProvider>
            </MemoryRouter>
        </SWRConfig>
    );
};

export const mswPathResponse = (
    server: SetupServerApi,
    path: string,
    json: object
) => {
    server.use(
        rest.get(path, (req, res, ctx) => {
            return res(ctx.json(json));
        })
    );
};
