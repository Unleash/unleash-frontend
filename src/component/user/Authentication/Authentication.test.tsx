import Authentication from 'component/user/Authentication/Authentication';
import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import {
    LOGIN_PASSWORD_ID,
    LOGIN_EMAIL_ID,
    LOGIN_BUTTON,
    AUTH_PAGE_ID,
    SSO_LOGIN_BUTTON,
} from 'utils/testIds';
import React from 'react';
import { TestContext, mswPathResponse } from 'utils/testUtils';

const server = setupServer();

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

afterEach(() => {
    server.resetHandlers();
});

test('should render password auth', async () => {
    mswPathResponse(server, '*', {
        defaultHidden: false,
        message: 'You must sign in in order to use Unleash',
        path: '/auth/simple/login',
        type: 'password',
        options: [],
    });

    render(
        <TestContext>
            <Authentication redirect="/" />
        </TestContext>
    );

    await screen.findByTestId(AUTH_PAGE_ID);
    await screen.findByTestId(LOGIN_PASSWORD_ID);
    await screen.findByTestId(LOGIN_EMAIL_ID);
    await screen.findByTestId(LOGIN_BUTTON);
    expect(document.body).toMatchSnapshot();
});

test('should not render password auth if defaultHidden is true', async () => {
    mswPathResponse(server, '*', {
        defaultHidden: true,
        message: 'You must sign in in order to use Unleash',
        path: '/auth/simple/login',
        type: 'password',
        options: [],
    });

    render(
        <TestContext>
            <Authentication redirect="/" />
        </TestContext>
    );

    await screen.findByTestId(AUTH_PAGE_ID);
    expect(screen.queryByTestId(LOGIN_EMAIL_ID)).toBeNull();
    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
    expect(screen.queryByTestId(LOGIN_BUTTON)).toBeNull();
    expect(document.body).toMatchSnapshot();
});

test('should render demo auth', async () => {
    mswPathResponse(server, '*', {
        defaultHidden: false,
        message: 'You must sign in in order to use Unleash',
        path: '/auth/demo/login',
        type: 'demo',
        options: [],
    });

    render(
        <TestContext>
            <Authentication redirect="/" />
        </TestContext>
    );

    await screen.findByTestId(AUTH_PAGE_ID);
    expect(screen.getByTestId(LOGIN_EMAIL_ID)).not.toBeNull();
    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
    expect(screen.getByTestId(LOGIN_BUTTON)).not.toBeNull();
    expect(document.body).toMatchSnapshot();
});

test('should render email auth', async () => {
    mswPathResponse(server, '*', {
        defaultHidden: false,
        message: 'You must sign in in order to use Unleash',
        path: '/auth/unsecure/login',
        type: 'unsecure',
        options: [],
    });

    render(
        <TestContext>
            <Authentication redirect="/" />
        </TestContext>
    );

    await screen.findByTestId(AUTH_PAGE_ID);
    expect(screen.getByTestId(LOGIN_EMAIL_ID)).not.toBeNull();
    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
    expect(screen.getByTestId(LOGIN_BUTTON)).not.toBeNull();
    expect(document.body).toMatchSnapshot();
});

test('should render Google auth', async () => {
    await testSSOAuthOption('google');
});

test('should render OIDC auth', async () => {
    await testSSOAuthOption('oidc');
});

test('should render SAML auth', async () => {
    await testSSOAuthOption('saml');
});

const testSSOAuthOption = async (authOption: string) => {
    const path = `/auth/${authOption}/login`;
    const testId = `${SSO_LOGIN_BUTTON}-${authOption}`;

    mswPathResponse(server, '*', {
        defaultHidden: true,
        message: 'You must sign in in order to use Unleash',
        options: [{ type: authOption, message: '...', path: path }],
        path: '/auth/simple/login',
        type: 'password',
    });

    render(
        <TestContext>
            <Authentication redirect="/" />
        </TestContext>
    );

    const ssoLink = await screen.findByTestId(testId);
    expect(ssoLink.getAttribute('href')).toEqual(path);
    expect(screen.queryByTestId(LOGIN_EMAIL_ID)).toBeNull();
    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
    expect(screen.queryByTestId(LOGIN_BUTTON)).toBeNull();
    expect(document.body).toMatchSnapshot();
};
