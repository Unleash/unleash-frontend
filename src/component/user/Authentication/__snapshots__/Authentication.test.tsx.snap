// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render password auth if defaultHidden is true 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
    </div>
  </div>
</body>
`;

exports[`should render Google auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <div
        class="makeStyles-flexColumn-80 makeStyles-contentSpacingY-72"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-loading="true"
          data-testid="SSO_LOGIN_BUTTON-google"
          href="/auth/google/login"
          style="height: 40px; color: rgb(0, 0, 0);"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall"
            >
              <div
                style="height: 35px; width: 35px;"
              />
            </span>
            ...
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`should render OIDC auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <div
        class="makeStyles-flexColumn-108 makeStyles-contentSpacingY-100"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-loading="true"
          data-testid="SSO_LOGIN_BUTTON-oidc"
          href="/auth/oidc/login"
          style="height: 40px; color: rgb(0, 0, 0);"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="height: 25px; width: 25px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM9 8V6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9z"
                />
              </svg>
            </span>
            ...
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`should render SAML auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <div
        class="makeStyles-flexColumn-136 makeStyles-contentSpacingY-128"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          data-loading="true"
          data-testid="SSO_LOGIN_BUTTON-saml"
          href="/auth/saml/login"
          style="height: 40px; color: rgb(0, 0, 0);"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="height: 25px; width: 25px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM9 8V6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9z"
                />
              </svg>
            </span>
            ...
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`should render demo auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <form>
        <div
          class="logo"
        />
        <div
          class="container"
        >
          <h2>
            Access the Unleash demo instance
          </h2>
          <p>
            No further data or Credit Card required
          </p>
          <div
            class="form"
          >
            <div
              class="MuiFormControl-root MuiTextField-root emailField"
              data-testid="LOGIN_EMAIL_ID"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Email
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  data-testid="email-input-field"
                  name="email"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-64 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-66"
                  >
                    <span>
                      Email
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-containedPrimary"
              data-testid="LOGIN_BUTTON"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <p>
            By accessing our demo instance, you agree to the Unleash 
            <a
              href="https://www.unleash-hosted.com/tos/"
              rel="noreferrer"
              target="_blank"
            >
              Customer Terms of Service
            </a>
             
            and 
            <a
              href="https://www.unleash-hosted.com/privacy-policy/"
              rel="noreferrer"
              target="_blank"
            >
              Privacy Policy
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</body>
`;

exports[`should render email auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <form>
        <div
          class="container"
        >
          <p>
            You must sign in in order to use Unleash
          </p>
          <p>
            This instance of Unleash is not set up with a secure authentication provider. You can read more about
             
            <a
              href="https://github.com/Unleash/unleash/blob/master/docs/securing-unleash.md"
              rel="noreferrer"
              target="_blank"
            >
              securing Unleash on GitHub
            </a>
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root"
            data-testid="LOGIN_EMAIL_ID"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              Email
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                data-testid="email-input-field"
                name="email"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-68 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-70"
                >
                  <span>
                    Email
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-containedPrimary"
              data-testid="LOGIN_BUTTON"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;

exports[`should render password auth 1`] = `
<body>
  <div>
    <div
      data-testid="AUTH_PAGE_ID"
    >
      <div
        style="max-width: 350px;"
      />
      <form>
        <div
          class="makeStyles-contentContainer-27 makeStyles-contentSpacingY-1"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            data-testid="LOGIN_EMAIL_ID"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
            >
              Username or email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                autocomplete="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                name="username"
                type="string"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-29 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-31"
                >
                  <span>
                    Username or email
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
            data-testid="LOGIN_PASSWORD_ID"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              style="padding-right: 0px;"
            >
              <input
                aria-invalid="false"
                autocomplete="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                name="password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-29 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-31"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="LOGIN_BUTTON"
            style="width: 150px; margin: 1rem auto;"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign in
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
      <div
        class="makeStyles-container-33"
      >
        <a
          class="makeStyles-link-34"
          href="/forgotten-password"
        >
          <p
            class="MuiTypography-root makeStyles-text-35 MuiTypography-body2"
          >
            Forgot password?
          </p>
        </a>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Don't have an account?
           
          <a
            class="makeStyles-link-34"
            href="https://www.getunleash.io/plans"
            rel="noopener noreferrer"
            target="_blank"
          >
            Sign up
          </a>
        </p>
      </div>
    </div>
  </div>
</body>
`;
